src/components/NavButtonWithTooltip/index.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useState } from "react";
+import { Tooltip, TooltipProps } from "antd";
+
+import {
+    NAV_BAR_TOOLTIP_OFFSET,
+    TOOLTIP_COLOR,
+    TOOLTIP_DELAY,
+} from "../../constants";
+import NavButton, { NavButtonProps } from "../NavButton";
+
+interface TooltipText {
+    default: string;
+    disabled?: string;
+}
+interface NavButtonWithTooltipProps extends NavButtonProps {
+    tooltipText?: TooltipText;
+    tooltipPlacement?: TooltipProps["placement"];
+}
+
+const NavButtonWithTooltip: React.FC<NavButtonWithTooltipProps> = ({
+    tooltipText = { default: "" },
+    titleText,
+    tooltipPlacement,
+    buttonType,
+    icon,
+    clickHandler,
+    isDisabled = false,
+    ...props
+}) => {
+    const getTooltipRenderText = () => {
+        return isDisabled && tooltipText.disabled
```

Comment:
nit: you could also have a default disabled tooltipText like "disabled" that would avoid the need for this double check. In general I would assume that if `isDisabled` is true it all has `tooltipText.disabled`